File: go/string/392_is_subsequence.go
package string

// isSubsequence 判断子序列, s 是 t 的子序列
func isSubsequence(s string, t string) bool {
	// 双指针
	idx1, idx2 := 0, 0
	m, n := len(s), len(t)

	for idx1 < m && idx2 < n {
		if s[idx1] == t[idx2] {
			idx1 += 1
		}
		idx2 += 1
	}
	return idx1 == m
}

File: rust/src/string/392_is_subsequence.rs


pub fn is_subsequence(s: String, t: String) -> bool {
    // s 是否 是 t 的子序列
    let (s, t) = (s.as_bytes(), t.as_bytes());
    let (mut idx1, mut idx2) = (0, 0);
    let (m, n) = (s.len(), t.len());

    while idx1 < m && idx2 < n {
        if s[idx1] == t[idx2] {
            idx1 += 1;
        }
        idx2 += 1;
    }
    idx1 == m
}
